container/rkt/handler.go

```diff
@@ -0,0 +1,488 @@
+	}
+
+	if parsed.Container == "" {
+		isPod = true
+		aliases = append(aliases, parsed.Pod)
+	} else {
+		aliases = append(aliases, parsed.Pod+":"+parsed.Container)
+	}
+
+	cgroupPaths := common.MakeCgroupPaths(cgroupSubsystems.MountPoints, name)
+
+	cHints, err := common.GetContainerHintsFromFile(*common.ArgContainerHints)
+	if err != nil {
+		return nil, err
+	}
+
+	// Generate the equivalent cgroup manager for this container.
+	cgroupManager := &cgroupfs.Manager{
+		Cgroups: &configs.Cgroup{
+			Name: name,
+		},
+		Paths: cgroupPaths,
+	}
+
+	pid := os.Getpid()
+	hasNetwork := false
+	if isPod {
+		hasNetwork = true
+
+		pids, err := cgroups.GetPids(cgroupPaths["cpu"])
+		if err == nil {
+			pid = pids[0]
+		} else {
+			glog.Infof("couldn't get processes as %v", err)
+		}
+	}
+
+	//SJP: unsure the point of this code, if it event does anything today?
+	var externalMounts []common.Mount
+	for _, container := range cHints.AllHosts {
+		if name == container.FullName {
+			/*libcontainerState.NetworkState = network.NetworkState{
+				VethHost:  container.NetworkInterface.VethHost,
+				VethChild: container.NetworkInterface.VethChild,
+			}
+			hasNetwork = true*/
+			externalMounts = container.Mounts
+			break
+		}
+	}
+
+	rootfsStorageDir := GetRootFs(rktPath, parsed)
+
+	handler := &rktContainerHandler{
+		name:               name,
+		cgroupSubsystems:   cgroupSubsystems,
+		machineInfoFactory: machineInfoFactory,
+		stopWatcher:        make(chan error),
+		cgroupPaths:        cgroupPaths,
+		cgroupManager:      cgroupManager,
+		fsInfo:             fsInfo,
+		hasNetwork:         hasNetwork,
+		externalMounts:     externalMounts,
+		watcher:            watcher,
+		rootFs:             rootFs,
+		isPod:              isPod,
+		aliases:            aliases,
+		pid:                pid,
+		rootfsStorageDir:   rootfsStorageDir,
+		ignoreMetrics:      ignoreMetrics,
+	}
+
+	if !ignoreMetrics.Has(container.DiskUsageMetrics) {
+		handler.fsHandler = common.NewFsHandler(time.Minute, rootfsStorageDir, "", fsInfo)
+	}
+
+	return handler, nil
+}
+
+func (self *rktContainerHandler) ContainerReference() (info.ContainerReference, error) {
+	return info.ContainerReference{
+		Name:      self.name,
+		Aliases:   self.aliases,
+		Namespace: RktNamespace,
+	}, nil
+}
+
+//SJP: Should a Rkt containe have have htis?
+func (self *rktContainerHandler) GetRootNetworkDevices() ([]info.NetInfo, error) {
+	nd := []info.NetInfo{}
+	return nd, nil
+}
+
+func (self *rktContainerHandler) Start() {
+	self.fsHandler.Start()
+}
+
+func (self *rktContainerHandler) Cleanup() {
+	self.fsHandler.Stop()
+}
+
+func (self *rktContainerHandler) GetSpec() (info.ContainerSpec, error) {
+	return common.GetSpec(self)
+}
+
+func (self *rktContainerHandler) getFsStats(stats *info.ContainerStats) error {
+	if self.ignoreMetrics.Has(container.DiskUsageMetrics) {
+		return nil
+	}
+
+	if self.isPod {
```

Comment:
Don't have that done yet, but yes, need to figure out how to role that up (assuming this is the journald stuff we talked about previously?)